### ISSUE TITLE
Humanoids getting stuck inside rooms

### ISSUE THREAD

**Issue by SlightPersimmon1** [CorsixTH/CorsixTH#2305]
Hello

This is my first full playthrough of TH using CorsixTH. I had a lot of fun, but encountered the same issue described here on two levels. Unfortunately one of them was the last one, when i was almost finishing the game.

What happens is that the humanoids get stuck inside the rooms, or even exiting/entering them. When this happens, i can no longer interact with them without the game throwing an exception and also i can't destroy the room.   

This is probably related to other issue reported here.

Here are 3 save files of the last level where i had 2 or 3 instances of the same problem. The reason i had so many is that i try to proceed (try to recover from the exception). Sometimes i can, sometimes i can't and the game crashes.   

[Saves.zip](https://github.com/CorsixTH/CorsixTH/files/10679700/Saves.zip)

Some screenshots of where the issue exists in the save files. 

![ss1](https://user-images.githubusercontent.com/65872044/217375785-b89a9557-2d1b-485b-80fe-4100607af37c.png)
![ss2](https://user-images.githubusercontent.com/65872044/217375787-047b66d0-32a6-4db1-bfdf-43ac28f37ce2.png)

I know this is really difficult to fix, but it would be great for when this happens, have a more graceful crash recovery, like for instance trying to remove the bugged actors and the room (if needed) from the code. 

Thank you again for all the work you have put into CTH! I really appreciate. And doing this with lua is even more impressive (you masochists).

**Comment by lewri** [comment]
I think this may be a blocking off areas bug. 

In the config.txt file (on windows it's in `C:\Users\<<user>>\Appdata\Roaming\CorsixTH`) there's a section for the blocking off areas setting:

```
-------------------------------------------------------------------------------
-- If true, parts of the hospital can be made inaccessible by blocking the path
-- with rooms or objects. If false, all parts of the hospital must be kept
-- accessible, the game will disallow any attempt to blocking the path.
--
allow_blocking_off_areas = true
```

If that value is set to true, change it to false. Likely any broken room won't be fixable. However, if you have a save from before the room broke you can remove the offending room and build it again. It will then with that config setting stop you putting objects down that block paths.

**Comment by SlightPersimmon1** [comment]
Thank you for your suggestion @lewri. Unfortunately i already have "allow_blocking_off_areas = false" . 

**_Edit: On all saves the zoom handler appears to have crashed, likely due to other things. Did you have a save from before one of these crashes?_**
 
Now this is an interesting observation. It may or may not be related, but i am using the D3D9.dll dgVoodoo wrapper to upscale the game window with a nearest neighbor algorithm - force point sampled, to avoid the blurry screen due the default game scaling (linear scaling, i think?).

My native screen resolution is 1680x1050 and i though that i setup the config file to play at 960x600. 

```
width = 960
height = 600
```
BUT, and this is the interesting part, i think i messed up the height (due probably a typo) and was actually playing at 
960x581  
   
```
width = 960
height = 581
```
No idea where that 581 came from. 

I corrected that and i will see if the issue persists. Unfortunately i don't have a previous save for that campaign, but i started playing the "ChrizmanTV and beyond" campaign and will post here if the issue occurs again. 

Thanks again.

**Comment by lewri** [comment]
> Now this is an interesting observation. It may or may not be related, but i am using the D3D9.dll dgVoodoo wrapper to upscale the game window with a nearest neighbor algorithm - force point sampled, to avoid the blurry screen due the default game scaling (linear scaling, i think?).


That... may be interesting regarding #2287 
I'm wondering though if using dgVoodoo means the game depends on that DLL, and as a result of it missing the game won't zoom? Bit of a swing in the dark for me though.

**Comment by mugmuggy** [comment]
So addressing cardio gram is because we've had a patient dying enter the room, same as #2278, #1584 and others. Maybe caused by the door/bench position if they were seated before entering the room, but either way they should die if they are dying. The subsequent errors are probably as there is still a relationship between the crashed patient and the room and the subsequent idle checks in finding staff members take out the other doctors/rooms, with the crashed patients empty action queue.

**Comment by Matvey1705** [comment]
Yeah, it happens to me often too, especially when here alot patients in the hospital

**Comment by lewri** [comment]
@mugmuggy was this one meant to be closed by #2310 ?

**Comment by lewri** [comment]
Fixed by #2310
